se_ompl/include/se/path_planner_ompl.hpp

```diff
@@ -0,0 +1,519 @@
+   * @param [in] min_boundary Lower boundaries in x, y and z of the map.
+   * @param [in] max_boundary Upper boundaries in x, y and z of the map.
+   * @param [out] space The ompl space to set the boundaries.
+   */
+
+  void setSpaceBoundaries();
+  void setSpaceBoundaries_m();
+
+  void reduceToControlPointCorridorRadius(Path_v::Ptr path_v);
+
+  Path_v::Ptr path_v_ = nullptr;
+  Path_v::Ptr path_not_simplified_v_ = nullptr;
+
+  std::shared_ptr<Octree<FieldType> > octree_ptr_ = nullptr;
+  std::shared_ptr<CollisionCheckerV<FieldType> > pcc_ = nullptr;
+  const Planning_Configuration planning_params_;
+  float solving_time_;
+  og::SimpleSetupPtr ss_; /// Create the set of classes typically needed to solve a
+
+
+  int safety_radius_v_;
+  PlannerType planner_type_;
+
+  Path<kDim>::Ptr path_ = nullptr;
+  Path<kDim>::Ptr path_not_simplified_ = nullptr;
+
+
+  // OmplSetup<FieldType> ompl_setup_;
+
+
+  bool start_end_occupied_ = false;
+  bool ompl_failed_ = false;
+
+  // planing bounds, if set
+  Eigen::Vector3i lower_bound_v_;
+  Eigen::Vector3i upper_bound_v_;
+
+  Eigen::Vector3f upper_bound_;
+  Eigen::Vector3f lower_bound_;
+
+};
+
+template<typename FieldType>
+PathPlannerOmpl<FieldType>::PathPlannerOmpl(const std::shared_ptr<Octree<FieldType> > octree_ptr,
+                                            const std::shared_ptr<CollisionCheckerV<FieldType> > pcc,
+                                            const Planning_Configuration &planning_config)
+    :
+    octree_ptr_(octree_ptr),
+    pcc_(pcc),
+    planning_params_(planning_config),
+    solving_time_(planning_config.ompl_solving_time),
+    ss_(aligned_shared<og::SimpleSetup>(ob::StateSpacePtr(new ob::RealVectorStateSpace(3))))
+//    ss_(ob::StateSpacePtr(std::make_shared<ob::RealVectorStateSpace>(kDim))),
+{
+  safety_radius_v_ =
+      static_cast<int>(planning_config.robot_safety_radius / octree_ptr_->voxelDim());
+  // Contstruct optimizing planner using RRT algorithm
+  // set Planner
+  DLOG(INFO) << "safety_radius_v_" << safety_radius_v_;
+  planner_type_ = planning_config.planner_type;
+  path_ = aligned_shared<Path<kDim> >();
+  path_not_simplified_ = aligned_shared<Path<kDim> >();
+}
+
+/**
+ * IMPLEMENTATION
+ */
+
+
+
+
+template<typename FieldType>
+bool PathPlannerOmpl<FieldType>::setStartGoal(const Eigen::Vector3i &start_v,
+                                              const Eigen::Vector3i &goal_v) {
+
+  std::chrono::time_point<std::chrono::steady_clock> timings[3];
+  // double normal_time = std::chrono::duration_cast<std::chrono::duration<double> >(timings[2]
+  // -timings[1]).count();
+  ss_->clear();
+  if (!pcc_->isSphereSkeletonFree(start_v, safety_radius_v_)) {
+    LOG(INFO) << "\033[1;31mStart at " << start_v.format(InLine) << " is occupied "
+              << octree_ptr_->get(start_v).x << "\033[0m";
+
+    //LOG(ERROR) << "Start is occupied";
+    // start_end_occupied_ = true;
+    // return false;
```

Comment:
Should this be commented out?

Reply:
it should never be true, as it's always the current drone position. adding a fail safe